sensors/Pms.py

```diff
@@ -0,0 +1,234 @@
+        return μgm3_to_gpl(self.__reading[DATA_INDICES['PM2.5']])
+    
+    
+    
+    #greater than 
+    def get_gt_0_3_raw(self)->float:
+        return self.__reading[DATA_INDICES['gt0.3μm']]
+    
+    def get_gt_0_5_raw(self)->float:
+        return self.__reading[DATA_INDICES['gt0.5μm']]
+    
+    def get_gt_1_raw(self)->float:
+        return self.__reading[DATA_INDICES['gt1.0μm']]
+    
+    def get_gt_2_5_raw(self)->float:
+        return self.__reading[DATA_INDICES['gt2.5μm']]
+
+    def get_gt_5_raw(self)->float:
+        return self.__reading[DATA_INDICES['gt5.0μm']]
+    
+    def get_gt_10_raw(self)->float:
+        return self.__reading[DATA_INDICES['gt10μm']]
+    
+    # processed
+    def get_gt_0_3(self)->float:
+        return pms_gt_output_to_si(self.__reading[DATA_INDICES['gt0.3μm']])
+    
+    def get_gt_0_5(self)->float:
+        return pms_gt_output_to_si(self.__reading[DATA_INDICES['gt0.5μm']])
+    
+    def get_gt_1(self)->float:
+        return pms_gt_output_to_si(self.__reading[DATA_INDICES['gt1.0μm']])
+    
+    def get_gt_2_5(self)->float:
+        return pms_gt_output_to_si(self.__reading[DATA_INDICES['gt2.5μm']])
+
+    def get_gt_5(self)->float:
+        return pms_gt_output_to_si(self.__reading[DATA_INDICES['gt5.0μm']])
+    
+    def get_gt_10(self)->float:
+        return pms_gt_output_to_si(self.__reading[DATA_INDICES['gt10μm']])
+    
+    
+
+	# Get raw data
+    def get_raw_data(self)-> float:
+        
+        ser = self.connect_to_sensor()
+        
+        start = time.time()
+
+        sof_index = 0
+
+        while True:
+            elapsed = time.time() - start
+            if elapsed > 5:
+                raise ReadTimeoutError("PMS5003 Read Timeout: Could not find start of frame")
+
+            sof = ser.read(1)
+            if len(sof) == 0:
+                raise SerialTimeoutError("PMS5003 Read Timeout: Failed to read start of frame byte")
+            sof = ord(sof) if type(sof) is bytes else sof
+
+            if sof == PMS5003_SOF[sof_index]:
+                if sof_index == 0:
+                    sof_index = 1
+                elif sof_index == 1:
+                    break
+            else:
+                sof_index = 0
+
+        checksum = sum(PMS5003_SOF)
+
+        data = bytearray(ser.read(2))  # Get frame length packet
+        if len(data) != 2:
+            raise SerialTimeoutError("PMS5003 Read Timeout: Could not find length packet")
+        checksum += sum(data)
+        frame_length = struct.unpack(">H", data)[0]
+
+        raw_data = bytearray(ser.read(frame_length))
+        print("frame_length:",frame_length)
+    
+
+        if len(raw_data) != frame_length:
+            raise SerialTimeoutError("PMS5003 Read Timeout: Invalid frame length. Got {} bytes, expected {}.".format(len(raw_data), frame_length))
+
+
+        return raw_data
+
+    # Take reading from the sensor
+    # (with retry, it's garunteed to get an output with up to 15 trials )
```

Comment:
Is this comment applicable to the PMS sensor?

Remove it if it is not

Reply:
Those were comments from DHT, removed.